### ISSUE TITLE
Questions about matching loss

### ISSUE THREAD

**Issue by EliotChenKJ** [vincent-thevenin/Realistic-Neural-Talking-Head-Models#27]
Hello~I have two questions about matching loss:

- In my opinion, the matrix W and the pose-ignorance code are acutually used in two ways. W is used as the projected embeddings of classes in projection cGAN, while the other serves as the "identity code". However, i don't actually understand **what it means that we choose to make Wi closer to the identity code,** and **how it'll affect the fine-tuning process**.

- the dataset used in fine-tuning: if we've meta-trained the network with large dataset and we would like to fine-tune it with one personal image (One-shot), does that means that we only have one training data in fine-tuning phase?  Then how can we train the network in cGAN way? Consisitently feeding this data into D as real data?

I will appreciate that if you can share your opinions with me~

**Comment by vincent-thevenin** [comment]
Hi @EliotChenKJ 

- The main reason you would want to make Wi closer to the embedding vector for person i is for fine-tuning. On its own W also represents identity code, it's contains person specific information just as the embedding vector. You can train the network without the matching loss, thus making Wi and ei different. But you won't be able to fine-tune after that by using the unseen e' to replace Wi. The author calls this feed-forward (FF).

- Yes you will only have that face for one-shot. If you're doing fine-tuning then you will use the whole network without the embedder., not just the discriminator.

**Comment by asterisk37n** [comment]
Is it a good idea to replace W_i with e' for whole dataset, making e as pose-ignorant embedding and identical person projection embedding ?

**Comment by EliotChenKJ** [comment]
Now i may understand it a little further.
- The "projection cGANs" method used in this paper is not exactly like the origin one :). In this paper, the writer was inspired by the pjGANs idea and came up with a W matrix, which was not so related to discriminator like the origin one did.
- Like the vincent siad, making W matrix mimic the identity-ignorance code intended to further train the G & D in fine-tuning phase. 
Thank for vincent's patient answer.
